artifactory/commands/transferfiles/filesdiff.go

```diff
@@ -171,13 +171,76 @@ func convertResultsToFileRepresentation(results []servicesUtils.ResultItem) (fil
+	// Handle repository with "Docker" package type
+	if f.packageType == docker {
+		return f.getDockerTimeFrameFilesDiff(repoKey, fromTimestamp, toTimestamp, paginationOffset)
+	}
+	//Handle all others types
 	query := generateDiffAqlQuery(repoKey, fromTimestamp, toTimestamp, paginationOffset)
 	return runAql(f.context, f.srcRtDetails, query)
 }
 
+// We handle docker repositories differently from other repositories.
+// The reason is: if a layout is already uploaded to "artifactory", and we try to upload it again to a different repository,
+// its modification date will be the date when it was last modified by "artifactory" (and not the upload date) - so we could miss this layout in "phase 2".
+// To solve this problem, we will look for all "manifest.json" and "list.manifest.json" files, and for each "manifest.json", we will run a search AQL in Artifactory
+// to get all artifacts in its path (that includes the "manifest.json" file itself and all its layouts).
+func (f *filesDiffPhase) getDockerTimeFrameFilesDiff(repoKey, fromTimestamp, toTimestamp string, paginationOffset int) (aqlResult *servicesUtils.AqlSearchResult, err error) {
+	// Get all modified manifest files ("manifest.json" and list.manifest.json" files)
+	query := generateDockerManifestAqlQuery(repoKey, fromTimestamp, toTimestamp, paginationOffset)
+	manifestFilesResult, err := runAql(f.context, f.srcRtDetails, query)
+	if err != nil {
+		return nil, err
+	}
+	var result []servicesUtils.ResultItem
+	var manifestPaths []string
+	// Add the "list.manifest.json" files to the result, skip "manifest.json" files and save their paths separately.
+	for _, file := range manifestFilesResult.Results {
+		if file.Name == "manifest.json" {
+			manifestPaths = append(manifestPaths, file.Path)
+		} else {
+			result = append(result, file)
+		}
+	}
+	// Get all content of directories containing a "manifest.json" file.
+	query = generateGetDirContentAqlQuery(repoKey, manifestPaths)
+	pathsResult, err := runAql(f.context, f.srcRtDetails, query)
+	if err != nil {
+		return nil, err
+	}
+	// Merge "list.manifest.json" files with all other files.
+	result = append(result, pathsResult.Results...)
+	aqlResult = &servicesUtils.AqlSearchResult{}
+	aqlResult.Results = result
+	return
+}
+
 func generateDiffAqlQuery(repoKey, fromTimestamp, toTimestamp string, paginationOffset int) string {
 	query := fmt.Sprintf(`items.find({"$and":[{"modified":{"$gte":"%s"}},{"modified":{"$lt":"%s"}},{"repo":"%s","path":{"$match":"*"},"name":{"$match":"*"}}]})`, fromTimestamp, toTimestamp, repoKey)
 	query += `.include("repo","path","name","modified")`
 	query += fmt.Sprintf(`.sort({"$asc":["modified"]}).offset(%d).limit(%d)`, paginationOffset*AqlPaginationLimit, AqlPaginationLimit)
 	return query
 }
+
+// This method generates an AQL that searches for all content in list of paths.
```

Comment:
method --> function
... for all content in list of paths. --> ... for all the content in the list list of provided Artifactory paths.